lib/magic_test/support.rb

```diff
@@ -102,5 +86,29 @@ def magic_test
         retry
       end
     end
+
+    private
+
+    def get_last
+      history_lines = Readline::HISTORY.to_a.last(20)
+      i = 2
+      last = history_lines.last(2).first
+      last_block = [last]
+      if last == 'end' || last.first(4) == 'end '
+        i += 1
+        last_block.unshift(history_lines.last(i).first)
+        until !last_block.first.match(/^(\s+)/)&[0] do
+          i += 1
+          last_block.unshift(history_lines.last(i).first)
+        end
+      end
+      return last_block
+    end
+
+    # TODO this feels like it's going to end up burning people who have other support files in `test` or `spec` that don't include `helper` in the name.
+    def get_last_caller(caller)
+      binding.pry
+      caller select { |s| s.include?("/test/") || s.include?("/spec/") }.reject { |s| s.include?("helper") }
```

Comment:
No caller.select required here? I didn't know that.

Reply:
must have been a typo